src/map_io/s2map.cc

```diff
@@ -58,27 +54,35 @@ struct S2MapDescrHeader {
 	char bulk[2290];  // unknown
 } /* size 2352 */;
 
+bool is_valid_cstring(const char* text, size_t max_len) {
+	for (size_t i = 0; i < max_len; ++i) {
+		if (text[i] == 0) {
+			return true;
+		}
+	}
+	return false;
+}
+
 // Some basic checks to identify obviously invalid headers
-bool is_valid_header(const S2MapDescrHeader& header) {
+void validate_header(const S2MapDescrHeader& header) {
```

Comment:
I did some tests copying some RTTR maps to a folder. but when I tried to open the folder in the editor it tries to preload all maps failing for one map with the player numbers set to 8 see check in line 80. 
Would it be possible to show the filenname of the failed map as well? 
Furthermore I believe we should allow 8 players now, to allow for such rttr map.